File: m4-h2/src/main/java/ru/yegorpilipenko/otus/spring/m4h2/repository/BookRepository.java
package ru.yegorpilipenko.otus.spring.m4h2.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import ru.yegorpilipenko.otus.spring.m4h2.dto.ShortBook;
import ru.yegorpilipenko.otus.spring.m4h2.model.Book;

import java.util.Collection;

public interface BookRepository extends MongoRepository<Book,String>, ExtendedBookRepository {

    @Query("{}")
    Collection<ShortBook> findAllShort();

    int removeById(String id);

}

File: m2-h3/src/main/kotlin/ru/yegorpilipenko/otus/spring/m2h3/repository/BookRepository.kt
package ru.yegorpilipenko.otus.spring.m2h3.repository

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import ru.yegorpilipenko.otus.spring.m2h3.dto.ShortBook
import ru.yegorpilipenko.otus.spring.m2h3.model.Book

interface BookRepository: MongoRepository<Book, String> {

    @Query("{}")
    fun findAllShort(): Collection<ShortBook>

    fun removeById(id: String): Int

}
